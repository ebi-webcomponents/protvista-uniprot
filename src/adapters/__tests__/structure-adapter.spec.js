import {
  getAllFeatureStructures,
  mergeOverlappingIntervals,
} from '../structure-adapter';

import entryData from './__mocks__/uniprotkb-entry-data';

describe('structure data', () => {
  it('should turn structures into features', () => {
    const features = getAllFeatureStructures(entryData);
    expect(features).toMatchSnapshot();
  });

  it('should merge Overlapping Intervals', () => {
    const features = getAllFeatureStructures(entryData);
    const overlapping = mergeOverlappingIntervals(features);
    expect(overlapping).toMatchSnapshot();
  });
});
