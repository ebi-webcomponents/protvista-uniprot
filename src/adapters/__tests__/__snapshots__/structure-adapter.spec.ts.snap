// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`structure data should merge Overlapping Intervals 1`] = `
[
  {
    "category": "STRUCTURE_COVERAGE",
    "end": 0,
    "start": 0,
    "structures": [
      {
        "description": "Method: X-ray. Resolution: 1.50 A. ",
        "end": 0,
        "source": {
          "id": "1AAP",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AAP",
        },
        "start": 0,
      },
    ],
    "type": "PDBE_COVER",
  },
  {
    "category": "STRUCTURE_COVERAGE",
    "end": 711,
    "start": 672,
    "structures": [
      {
        "description": "Method: NMR. Chains: A=672-699. ",
        "end": 699,
        "source": {
          "id": "1AMB",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AMB",
        },
        "start": 672,
      },
      {
        "description": "Method: NMR. Chains: A=672-699. ",
        "end": 699,
        "source": {
          "id": "1AMC",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AMC",
        },
        "start": 672,
      },
      {
        "description": "Method: NMR. Chains: A=672-711. ",
        "end": 711,
        "source": {
          "id": "1AML",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AML",
        },
        "start": 672,
      },
      {
        "description": "Method: NMR. Chains: A=672-711. ",
        "end": 711,
        "source": {
          "id": "1BA4",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1BA4",
        },
        "start": 672,
      },
    ],
    "type": "PDBE_COVER",
  },
]
`;

exports[`structure data should turn structures into features 1`] = `
[
  {
    "category": "STRUCTURE_COVERAGE",
    "end": 0,
    "start": 0,
    "structures": [
      {
        "description": "Method: X-ray. Resolution: 1.50 A. ",
        "end": 0,
        "source": {
          "id": "1AAP",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AAP",
        },
        "start": 0,
      },
    ],
    "type": "PDBE_COVER",
  },
  {
    "category": "STRUCTURE_COVERAGE",
    "end": 699,
    "start": 672,
    "structures": [
      {
        "description": "Method: NMR. Chains: A=672-699. ",
        "end": 699,
        "source": {
          "id": "1AMB",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AMB",
        },
        "start": 672,
      },
    ],
    "type": "PDBE_COVER",
  },
  {
    "category": "STRUCTURE_COVERAGE",
    "end": 699,
    "start": 672,
    "structures": [
      {
        "description": "Method: NMR. Chains: A=672-699. ",
        "end": 699,
        "source": {
          "id": "1AMC",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AMC",
        },
        "start": 672,
      },
    ],
    "type": "PDBE_COVER",
  },
  {
    "category": "STRUCTURE_COVERAGE",
    "end": 711,
    "start": 672,
    "structures": [
      {
        "description": "Method: NMR. Chains: A=672-711. ",
        "end": 711,
        "source": {
          "id": "1AML",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AML",
        },
        "start": 672,
      },
    ],
    "type": "PDBE_COVER",
  },
  {
    "category": "STRUCTURE_COVERAGE",
    "end": 711,
    "start": 672,
    "structures": [
      {
        "description": "Method: NMR. Chains: A=672-711. ",
        "end": 711,
        "source": {
          "id": "1BA4",
          "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1BA4",
        },
        "start": 672,
      },
    ],
    "type": "PDBE_COVER",
  },
]
`;
